### PR TITLE
Enables mssql support

### PR DESCRIPTION
**Pull request by btrepp**
Port of @dlob work in gogs (gogits/gogs#3772)
Enables options in index.js
Enables MSSQL as a database option in go.
Sets ID to 0 on initial migration. Required for MSSQL insert  #statements.

Signed-off-by: Beau Trepp <beautrepp@gmail.com>

### CODE REVIEW

**Comment by lunny** [comment]
LGTM.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
@btrepp  Maybe you have to `govendor add github.com/denisenkom/go-mssqldb`, since Gitea use govendor to manage dependencies.

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -93,6 +93,7 @@ func Migrate(x *xorm.Engine) error {
 	} else if !has {
 		// If the version record does not exist we think
 		// it is a fresh installation and we can skip all migrations.
+		currentVersion.ID= 0
```
Comment (tboerger) [inline models/migrations/migrations.go]: Missing space :)

**Review by tboerger — COMMENTED** [review]
Please also add the new dependency to the vendor directory via govendor, than it looks fine.

**Comment by btrepp** [comment]
Added a new commit with what is needed for govendor (never used that before, sorry).

Do you want me to fix up the missing space and squash the commits into one?.

**Comment by tboerger** [comment]
Please fix the format and keep the vendoring in a separate commit. Than I'm fine with that.

**Comment by andreynering** [comment]
Works surprisingly well.

LGTM, but should be merged only on 1.1.0

**Comment by tboerger** [comment]
There is also another remaining issue related to that, if I remember correctly on the explore page

**Pull request merged**